### DISCUSSION TITLE
Can I apply vector rotation to a String with a given font?

### DISCUSSION THREAD

**Discussion by lweier** [fonttools/fonttools#3365] [Q&A] [answered]
Hello,
I want to load a font using fonttools and write a given String in that font. I then want to rotate and draw it, but I want to avoid rotating the rasterized image, so I would like to apply vector rotation before drawing it.

Is that possible? If so, how?

**Comment by behdad** [comment]
Check out the `TransformPen` from the `fontTools.pens.transformPen` module.

**Reply by lweier** [reply]
First off: Thanks for the response!
I am a little confused with the pens. I want to generate png-images at the end. Preferably entire sentences. Is that possible with the pen? I couldn't find any coding examples.

**Reply by behdad** [reply, marked as answer]
Fonttools itself doesn't have facilities to write to PNG. Looks like you might be better off with a library like pangocairo.
